devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,606 @@
+	if (encapsulatedCommand) {
+		log.debug "Parsed Crc16Encap into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract CRC16 command from $cmd"
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using CRC16 Encapsulation, command: $cmd"
+	zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private encapSequence(cmds, Integer delay = 250) {
+	delayBetween(cmds.collect { encap(it) }, delay)
+}
+
+private getParameterMap() {
+	[
+		[
+			name           : "Dimming Speed", key: "dimmingSpeed", type: "range",
+			parameterNumber: 1, size: 1, defaultValue: 3,
+			range          : "1..100",
+			description    : "How fast or slow the light turns on when you hold the switch (ie: dimming from 10-20%, 80-60%, etc). Example of how the values work: 0 - Instant On, 1 = 1 second, 100 = 100 seconds. This parameter can be set without a HUB from the Configuration Button. Finally, if you are using a,dumb switch in a 3-Way setting, this parameter will not work if you manually press the dumb switch (it will only work if you press the smart switch)."
+		],
+		[
+			name           : "Default Level (Z-Wave)", key: "defaultLevel(Z-Wave)", type: "range",
+			parameterNumber: 10, size: 1, defaultValue: 0,
+			range          : "0..100",
+			description    : "Default dim level for the switch when powered on via a Z-Wave command. This is useful if you'd like your switch to turn on to a certain level when remotely controlling the switch during certain times. For example, you could have your switch only turn on to 10% brightness during the hours of 10pm - 6am when you remotely control it. Example of how the values work: 0 = Switch will return to level it was prior to being off, 1 = 1%, 100 = 100%. This parameter can be set without a HUB from the Configuration Button."
+		],
+		[
+			name           : "Power On State", key: "powerOnState", type: "range",
+			parameterNumber: 11, size: 1, defaultValue: 0,
+			range          : "0..101",
+			description    : "When power is restored, the switch reverts to either On, Off, or Last Level. Example of how the values work: 0 = Off, 1-100 = Specific % On, 101 = Returns to Level before Power Outage. This parameter can be set without a HUB from the Configuration Button."
+		],
+		[
+			name           : "Association Behavior", key: "associationBehavior", type: "enum",
```

Comment:
I question whether we should include this, since you're only able to set association with another controller installed or by using a custom DTH.

Reply:
I deleted this preference